src/openssl/ssl/socket.cr

```diff
@@ -235,4 +240,17 @@ abstract class OpenSSL::SSL::Socket < IO
       raise NotImplementedError.new("#{io.class}#write_timeout=")
     end
   end
+
+  # Returns the `OpenSSL::X509::Certificate` the peer presented, if a
+  # connection was esablished.
+  #
+  # NOTE: Due to the protocol definition, a TLS/SSL server will always send a
+  # certificate, if present. A client will only send a certificate when
+  # explicitly requested to do so by the server (see SSL_CTX_set_verify(3)). If
+  # an anonymous cipher is used, no certificates are sent. That a certificate
+  # is returned does not indicate information about the verification state.
+  def peer_certificate
```

Comment:
Since the return type is nilable wouldn't be moar apt to name it `peer_certificate?`?.

Reply:
Not necessarily, but might be a good idea.

In any case, please specify the return type in the methods.

Reply:
Adding a `?` is meant to denote a nilable _variant_ of a raising method. There is no raising `peer_certificate` method, so the method is simply returns a nilable, and shouldn't have a `?` suffix.

Reply:
Oh I see, the other `#peer_certificate` is in a subclass. Then no `?`.

Reply:
We should start to write a style guide with such idioms, they're only said here and there. It would be nice to collect them.

Reply:
I'm not sure if I followed this thread correctly. I think if I just add the return type explicitly then it'll be good?